src/Pages/SubsitesVirtualPage.php

```diff
@@ -21,139 +19,68 @@
-        }
-        $fields->replaceField('CopyContentFromID', $pageSelectionField);
-
-        // Create links back to the original object in the CMS
-        if ($this->CopyContentFromID) {
-            $editLink = Controller::join_links(
-                CMSPageEditController::singleton()->Link('show'),
-                $this->CopyContentFromID
-            );
+        $this->beforeUpdateCMSFields(function (FieldList $fields) {
+            $subsites = DataObject::get(Subsite::class);
+            if (!$subsites) {
+                $subsites = ArrayList::create();
+            } else {
+                $subsites = ArrayList::create($subsites->toArray());
+            }
+            $subsites->push(ArrayData::create(['Title' => 'Main site', 'ID' => 0]));
 
-            $linkToContent = "
-				<a class=\"cmsEditlink\" href=\"$editLink\">" .
-                _t('SilverStripe\\CMS\\Model\\VirtualPage.EDITCONTENT', 'Click here to edit the content') .
-                '</a>';
-            $fields->removeByName('VirtualPageContentLinkLabel');
             $fields->addFieldToTab(
                 'Root.Main',
-                $linkToContentLabelField = LiteralField::create('VirtualPageContentLinkLabel', $linkToContent),
-                'Title'
+                DropdownField::create(
+                    'CopyContentFromID_SubsiteID',
+                    _t(__CLASS__ . '.SubsiteField', 'Subsite'),
+                    $subsites->map('ID', 'Title')
+                )->addExtraClass('subsitestreedropdownfield-chooser no-change-track'),
+                'CopyContentFromID'
             );
-        }
 
-
-        $fields->addFieldToTab(
-            'Root.Main',
-            TextField::create(
-                'CustomMetaTitle',
-                $this->fieldLabel('CustomMetaTitle')
-            )->setDescription(_t(__CLASS__ . '.OverrideNote', 'Overrides inherited value from the source')),
-            'MetaTitle'
-        );
-        $fields->addFieldToTab(
-            'Root.Main',
-            TextareaField::create(
-                'CustomMetaKeywords',
-                $this->fieldLabel('CustomMetaKeywords')
-            )->setDescription(_t(__CLASS__ . '.OverrideNote', 'Overrides inherited value from the source')),
-            'MetaKeywords'
-        );
-        $fields->addFieldToTab(
-            'Root.Main',
-            TextareaField::create(
-                'CustomMetaDescription',
-                $this->fieldLabel('CustomMetaDescription')
-            )->setDescription(_t(__CLASS__ . '.OverrideNote', 'Overrides inherited value from the source')),
-            'MetaDescription'
-        );
-        $fields->addFieldToTab(
-            'Root.Main',
-            TextField::create(
-                'CustomExtraMeta',
-                $this->fieldLabel('CustomExtraMeta')
-            )->setDescription(_t(__CLASS__ . '.OverrideNote', 'Overrides inherited value from the source')),
-            'ExtraMeta'
-        );
```

Comment:
Moved the relevant ones to `VirtualPage` (see https://github.com/silverstripe/silverstripe-cms/pull/2983)

Removed `CustomMetaTitle` and `CustomMetaKeywords` because the fields they're intended to override were long-since removed from `SiteTree`.